packages/grafana-data/src/types/annotations.ts

```diff
@@ -0,0 +1,87 @@
+import { DataQuery, QueryEditorProps } from './datasource';
+import { DataFrame } from './dataFrame';
+import { ComponentType } from 'react';
+
+/**
+ * This JSON object is stored in the dashboard json model.
+ */
+export interface AnnotationQuery<TQuery extends DataQuery = DataQuery> {
+  datasource: string;
+  enable: boolean;
+  name: string;
+  iconColor: string;
+
+  // Standard datasource query
+  target?: TQuery;
+
+  // Convert a dataframe to an AnnotationEvent
+  mappings?: AnnotationEventMappings;
+}
+
+export interface AnnotationEvent {
+  id?: string;
+  annotation?: any;
+  dashboardId?: number;
+  panelId?: number;
+  userId?: number;
+  login?: string;
+  email?: string;
+  avatarUrl?: string;
+  time?: number;
+  timeEnd?: number;
+  isRegion?: boolean;
+  title?: string;
+  text?: string;
+  type?: string;
+  tags?: string[];
+
+  // Currently used to merge annotations from alerts and dashboard
+  source?: any; // source.type === 'dashboard'
+}
+
+/**
+ * @alpha -- any value other than `field` is experimental
+ */
+export enum AnnotationEventFieldSource {
+  Field = 'field', // Default -- find the value with a matching key
+  Text = 'text', // Write a constant string into the value
+  Skip = 'skip', // Do not include the field
+}
+
+export interface AnnotationEventFieldMapping {
```

Comment:
currently, we are only using the `field` structure, but I want to make sure we can also support setting the text directly and with template variables.